src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -133,6 +139,23 @@ public void setTask(Task target, Task editedTask) {
         requireAllNonNull(target, editedTask);
         taskList.setTask(target, editedTask);
         this.sortList();
+        try {
+			notifyObservers();
+		} catch (CommandException e) {
+			e.printStackTrace();
+		}
+    }
+
+    // =========== Subject Methods for Observer
+    // ================================================================================
+    public void notifyObservers() throws CommandException {
+        for (Observer observer : observers) {
+            observer.update();
```

Comment:
Your observer just saves the Task list?

Reply:
Somewhere else handles it already, and you're having the observer notified whenever we edit a task?

Reply:
actually yes HAHA